wordpress-mobile/WordPress-Android
# Site Creation Visual Cue for Unavailable or Invalid Domains

Fixes #10089 
Now when the user navigates to the 3rd step of the Site Creation process and they are selecting their domain they will be notified if the query they entered is already in use or invalid. Previously, only suggestions would have been shown. 

To test:
1. My Site. 
2. Switch site.
3. Press the add button.
4. Create a new WordPress.com site. 
5. Skip to the screen where you can search for a domain and begin utilizing different search queries to see the behavior. 

Below is a gif of this in action. 

<img src="https://user-images.githubusercontent.com/1509205/67918998-dffce780-fb5b-11e9-8bfc-25afceb49a91.gif" height="640">

Before on the left and the after on the right.
<img src="https://user-images.githubusercontent.com/1509205/67919295-0b340680-fb5d-11e9-8d04-582757ff46aa.png" height="640"> <img src="https://user-images.githubusercontent.com/1509205/67919312-1b4be600-fb5d-11e9-8205-4dab6fbf0625.png" height="640"> 

<img src="https://user-images.githubusercontent.com/1509205/67919331-30c11000-fb5d-11e9-8ca7-cb59b6123f9e.png" height="640"> <img src="https://user-images.githubusercontent.com/1509205/67919345-3ae30e80-fb5d-11e9-9cbd-c13f87a58bdb.png" height="640">

### Todo

- [ ] Do accessibility audit on the layout and make changes to support it. 
- [ ] Verify the domain cases that the app needs to validate.

PR submission checklist:
